runtimes/go/storage/objects/internal/providers/s3/bucket.go

```diff
@@ -151,6 +154,26 @@ func (b *bucket) Attrs(data types.AttrsData) (*types.ObjectAttrs, error) {
 	}, nil
 }
 
+func (b *bucket) GetUploadUrl(data types.UploadUrlData) (string, error) {
+	object := string(data.Object)
+	params := s3.PutObjectInput{
+		Bucket: &b.cfg.CloudName,
+		Key:    &object,
+	}
+	sign_opts := func(opts *s3.PresignOptions) {
+		opts.Expires = time.Duration(data.Ttl) * time.Second
+	}
+	req, err := b.presign_client.PresignPutObject(data.Ctx, &params, sign_opts)
```

Comment:
Do the error check here instead
```suggestion
	req, err := b.presign_client.PresignPutObject(data.Ctx, &params, sign_opts)
	if err != nil {
	    return "", mapErr(err)
	}
	return req.URL, nil
```

Reply:
Done